src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
-				{
-					max.z -= thickness;
-				}
+			//Set global element states.
 
-				if( facades.containsKey( EnumFacing.NORTH ) )
+			//calculate the side mask.
+			int facadeMask = 0;
+			for( Entry<EnumFacing, FacadeRenderState> ent : facadeStates.entrySet() )
+			{
+				EnumFacing s = ent.getKey();
+				if( s.getAxis() != side.getAxis() )
 				{
-					min.z += thickness;
+					FacadeRenderState otherState = ent.getValue();
+					if( !otherState.isTransparent() )
+					{
+						facadeMask |= 1 << s.ordinal();
+					}
 				}
 			}
+			//Setup the edge stripper.
+			edgeStripper.setBounds( fullBounds );
+			edgeStripper.setMask( facadeMask );
 
-			builder.addCube( min.x, min.y, min.z, max.x, max.y, max.z );
-		}
-		else
-		{
-			Vector3f busMin = new Vector3f( (float) busBounds.minX * 16, (float) busBounds.minY * 16, (float) busBounds.minZ * 16 );
-			Vector3f busMax = new Vector3f( (float) busBounds.maxX * 16, (float) busBounds.maxY * 16, (float) busBounds.maxZ * 16 );
+			//Setup the kicker.
+			kicker.setSide( sideIndex );
+			kicker.setFacadeMask( facadeMask );
+			kicker.setBox( fullBounds );
+			kicker.setThickness( thinFacades ? THIN_THICKNESS : THICK_THICKNESS );
 
-			if( side == EnumFacing.UP || side == EnumFacing.DOWN )
-			{
-				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, busMax.z, 16.0f, 16.0f, 16.0f );
-				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, 0.0f, 16.0f, 16.0f, busMin.z );
-				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, busMin.z, busMin.x, 16.0f, busMax.z );
-				this.renderSegmentBlockCurrentBounds( builder, min, max, busMax.x, 0.0f, busMin.z, 16.0f, 16.0f, busMax.z );
-			}
-			else if( side == EnumFacing.NORTH || side == EnumFacing.SOUTH )
+			for( BakedQuad quad : modelQuads )
 			{
-				if( facades.get( EnumFacing.UP ) != null )
+				//lookup the format in CachedFormat.
+				CachedFormat format = CachedFormat.lookup( quad.getFormat() );
+				//If this quad has a tint index, setup the tinter.
+				if( quad.hasTintIndex() )
 				{
-					max.y -= thickness;
+					tinter.setTint( blockColors.colorMultiplier( blockState, facadeAccess, pos, quad.getTintIndex() ) );
 				}
-
-				if( facades.get( EnumFacing.DOWN ) != null )
+				for( AxisAlignedBB box : holeStrips )
 				{
-					min.y += thickness;
+					//setup the clamper for this box
+					clamper.setClampBounds( box );
+					//Reset the pipeline, clears all enabled/disabled states.
+					pipeline.reset( format );
+					//Reset out collector.
+					collectorQuad.reset( format );
+					//Enable / disable the optional elements
+					pipeline.setElementState( "tinter", quad.hasTintIndex() );
+					pipeline.setElementState( "transparent", transparent );
+					//Prepare the pipeline for a quad.
+					pipeline.prepare( collectorQuad );
+
+					//Pipe our quad into the pipeline.
+					quad.pipe( pipeline );
+					//Check if the collector got any data.
+					if( collectorQuad.full )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)